scripts/bash_includes/tc/lua

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: BSD-2-Clause
+
+function download_lua {
+   local lua=$1
+   local version=$2
+   local filename=${lua}
+   local tarname="${lua}-${version}.tar.gz"
+   local url="https://www.lua.org/ftp/"
+   
+   download_file_in_cache "${url}" "${tarname}" lua
+   extract_cachefile_tar_gz "${tarname}" "${lua}-${version}" "${lua}"
+}
+
+all_funcs_list+=(build_lua)
+function build_lua {
+
+   pushd ${ARCH} 
+
+   lua="lua"
+   lua_version="5.4.1"
+   reset_cc_vars
+
+   show_work_on_component_msg "LUA"
+
+   if ! [ -d "${lua}" ]; then
+      echo "${TC}/${ARCH}/${lua} not found...downloading"
+      download_lua "${lua}" "${lua_version}"
+   fi
+
+   cd ${lua}
+
+   set_cc_vars_to_tc
+   echo "Bulding lua..."
+
+   make clean
+   
+   #build lua and luac static binaries
+   make generic CC="${CC} -std=gnu99" \
```

Comment:
Because `run_command2` is used everywhere, this should be turned into something like:
```
      local flags=""
      flags="$flags CC=\"$CC\" MYCFLAGS=\"-std=gnu99\""
      flags="$flags AR=\"$AR rcu\" RANLIB=\"$RANLIB\""
      flags="$flags MYLDFLAGS=\"-static\""
      run_command2 "make $flags" build.log
```